### PR TITLE
:lipstick: Assessment block redesign block redesign#645

### LINKED ISSUE #645: :art:  :lipstick:  Multi content Block Redesign: block redesign
## User stories

As a user, I want the "Multi content Block" component in the app to be redesigned to enhance its usability and visual appeal.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Technical Description
The current design is: 
image

It should be changed to:

The files can be found at `libs/features/convs-mgr/stories/blocks/library/multi-content-block/`. 

The designs are [here](https://www.figma.com/file/B6eRNqGu8W0JjW39dOI8ZN/Sprint-IV?type=design&node-id=358-50&mode=design&t=OS2KLCXDVLRb5Ky0-0)

### Acceptance Criteria

1. The block should look exactly like the designs.
2. Use of `rem`, `em` as opposed to `px`.
3. All changes should be isolated to the following  `libs/features/convs-mgr/stories/blocks/library/multi-content-block/`.

### PR DESCRIPTION
**Pull request by levy-web**
# Description

Re- designed assessment block UI. changed the Icon and added menu svg that toggles munu items to either display or hide

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes #645 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# Screenshot (optional)

Screenshot from 2023-11-03 07-53-12
Screenshot from 2023-11-03 07-53-17

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**: (optional)
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/assessment-brick/src/lib/components/assessment-brick/assessment-brick.component.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export class AssessmentBrickComponent implements OnInit, AfterViewInit, OnDestro
   /** Add JsPlumb connector to max score input */
   private _decorateInput()
   {
-    const inputs = document.getElementsByClassName('input-score-max');
+    const inputs = document.getElementsByClassName('option');
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/assessment-brick/src/lib/components/assessment-brick/assessment-brick.component.ts:67]: Any reason for this switch?

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html [hunk 2]
```diff
@@ -2,15 +2,21 @@
     <div [ngSwitch]="type">
         <div *ngIf="type !== endStoryAnchor" class="outer">
             <div class="block-title-container">
-                <i [class]="iconClass" class="icon-color"></i>
+                <i *ngIf="type != 31" [class]="iconClass" class="icon-color"></i>
+                <img *ngIf="type == 31" [src]="iconClass" />
                 <h4 class="block-title"> {{ blockTitle | transloco }}</h4>
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html:7]: Sync the latest changes there's a cleaner way to do this

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html [hunk 3]
```diff
@@ -2,15 +2,21 @@
...
             </div>
 
-            <div fxLayout="row" class="icon-block">
+            
+            <div *ngIf="isMenuOpen"  fxLayout="row" class="icon-block">
                 <span (click)="editBlock()" class="icons"><i class="far fa-edit"></i></span>
                 <span (click)="copyblock(block)" class="icons"><i class="far fa-clone"></i></span>
                 <span (click)="deleteBlock()" class="icons"><i class="far fa-trash-alt"></i></span>
             </div>
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/components/block/block.component.html:15]: Sync your branch to get the latest changes

File: libs/features/convs-mgr/stories/editor/src/lib/components/grouped-blocks/grouped-blocks.component.html [hunk 4]
```diff
@@ -2,7 +2,8 @@
   <ng-container *ngFor="let block of groupedBlocks">
     <div (click)="addBlock(block.type)" draggable="true" (dragend)="onDragEnd(block.type)" class="block">
       <div fxLayout="column" fxLayoutAlign="center center" class="blocks-main">
-        <i class="{{ block.blockIcon }}" id="icons"></i>
+        <i *ngIf="block.message !== 'Assessment'" class="{{ block.blockIcon }}" id="icons"></i>
+        <img *ngIf="block.message === 'Assessment'" [src]="block.blockIcon" />
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/editor/src/lib/components/grouped-blocks/grouped-blocks.component.html:6]: Use ngIf else with the else section in an ngtemplate to showcase this. (sync changes)

File: libs/features/convs-mgr/stories/blocks/library/assessment-brick/src/lib/components/assessment-brick/assessment-brick.component.ts (hunk 1 above)
Comment (levy-web) [reply libs/features/convs-mgr/stories/blocks/library/assessment-brick/src/lib/components/assessment-brick/assessment-brick.component.ts:67]: I added the option class with a maximum width so as to display the connectors at the end. when I use input-score-max connectors are at the center

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/assessment-brick.jsplumb.ts [hunk 5]
```diff
@@ -24,7 +24,7 @@ export function _AssessmentBrickDecoratePlumb(
     target: true,
     cssClass: 'block_endpoint',
     endpoint: 'Dot',
-    anchor: 'Left',
+    anchor: [0, 0.16 , 0, 1],
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/assessment-brick.jsplumb.ts:27]: Just have this as left for now.

File: libs/features/convs-mgr/stories/editor/src/lib/components/blocks-library/blocks-library.component.ts [hunk 6]
```diff
@@ -64,7 +64,7 @@ export class BlocksLibraryComponent implements OnInit, OnDestroy {
     { id: 'open-ended-question-block', type:StoryBlockTypes.OpenEndedQuestion, message: 'Open Ended Question', blockIcon:this.getBlockIcon(StoryBlockTypes.OpenEndedQuestion), blockCategory: 'questions-block' } as OpenEndedQuestionBlock,
     { id: 'keyword-jump-block', type:StoryBlockTypes.keyword, message: 'Keyword Jump', blockIcon:this.getBlockIcon(StoryBlockTypes.keyword), blockCategory: 'questions-block' } as KeywordMessageBlock,
     { id: 'event-block', type:StoryBlockTypes.Event, message: 'Event', blockIcon:this.getBlockIcon(StoryBlockTypes.Event), blockCategory: 'questions-block' } as EventBlock,
-    { id: 'assessment-brick', type:StoryBlockTypes.Assessment, message: 'Assessment', blockIcon:this.getBlockIcon(StoryBlockTypes.Assessment), blockCategory: 'bricks' } as AssessmentBrick,
+    { id: 'assessment-brick', type:StoryBlockTypes.Assessment, message: 'Assessment', svgIcon:this.getSvgIcon(StoryBlockTypes.Assessment), blockIcon:this.getBlockIcon(StoryBlockTypes.Assessment),  blockCategory: 'bricks' } as AssessmentBrick,
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/editor/src/lib/components/blocks-library/blocks-library.component.ts:67]: Use the blockIcon property so no need to change this.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Sync your changes and resolve conflicts

**Review by LemmyMwaura — APPROVED** [review]

**Pull request merged**